logger/mocklogger_test.go

```diff
@@ -0,0 +1,67 @@
+package logger
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	router "gopkg.in/Clever/kayvee-go.v5/router"
+)
+
+func TestMockLoggerImplementsKayveeLogger(t *testing.T) {
+	assert.Implements(t, (*KayveeLogger)(nil), &MockRouteCountLogger{}, "*MockRouteCountLogger should implement KayveeLogger")
+}
+
+func TestRouteCountsWithMockLogger(t *testing.T) {
+	routes := map[string](router.Rule){
+		"rule-one": router.Rule{
+			Matchers: router.RuleMatchers{
+				"foo": []string{"bar", "baz"},
+			},
+			Output: router.RuleOutput{
+				"out": "#-%{foo}-",
+			},
+		},
+		"rule-two": router.Rule{
+			Matchers: router.RuleMatchers{
+				"abc": []string{"def"},
+			},
+			Output: router.RuleOutput{
+				"more": "x",
+			},
+		},
+	}
+	testRouter := router.NewFromRoutes(routes)
+
+	mockLogger := NewMockCountLogger("testing")
```

Comment:
yep that sounds right. i was hoping users would adopt the dependency injecting path.

as you say, the problem with `thing.Logger = NewMockCountLogger()` is the initialization. if the `thing.Logger` is replaced before initialization occurs, the mock logger should setup everything just like the old `logger.Logger` would.